Source/WebCore/html/OffscreenCanvas.cpp

```diff
@@ -576,6 +577,20 @@ void OffscreenCanvas::reset()
     scheduleCommitToPlaceholderCanvas();
 }
 
+
+void OffscreenCanvas::queueTaskKeepingObjectAlive(TaskSource source, Function<void()>&& task)
+{
+    UNUSED_PARAM(source);
+    scriptExecutionContext()->postTask([protectedThis = Ref { *this }, task = WTFMove(task)] (ScriptExecutionContext&) mutable {
```

Comment:
This ain't right. You're not keeping the wrapper alive.

Reply:
I don't think we want to add this function at all. Use the one on ScriptExecutionContext instead.

Reply:
Please refer to https://github.com/WebKit/WebKit/blob/main/Introduction.md#js-wrapper-lifecycle-management

Reply:
Also relevant: https://github.com/WebKit/WebKit/blob/main/Introduction.md#creating-a-pending-activity

Reply:
Thanks for the review!

I'll add ActiveDOMObject to OffscreenCanvas, since it seems that's what's needed.

Reply:
Yeah, that would make most sense to me.